src/views/List/List.jsx

```diff
@@ -150,9 +156,31 @@ export default function List({ token }) {
               </p>
             ) : null}
             {copyOfData.map((listItem, index) => {
-              const { name, isActive } = listItem;
+              const {
+                name,
+                isActive,
+                frequency,
+                timesPurchased,
+                lastPurchasedAt,
+              } = listItem;
+              let buyIndicator = '';
+              let badge = '';
+              if (
+                timesPurchased === 1 ||
+                ((Date.now() - lastPurchasedAt) / 86400000 >= frequency * 2 &&
+                  lastPurchasedAt !== null)
+              ) {
+                buyIndicator = 'inactive';
+              } else if (frequency < 7) {
+                buyIndicator = 'soon';
+              } else if (frequency >= 7 && frequency <= 30) {
+                buyIndicator = 'kind-of-soon';
+              } else if (frequency > 30) {
+                buyIndicator = 'not-soon';
+              }
+              badge = buyIndicator.replace('-', ' ');
```

Comment:
.replace is only replacing the first hyphen just FYI so kind-of-soon is showing as kind of-soon 

I think you either don't need to replace hyphens, they look ok in the browser or maybe try replaceAll()?

looks like it would work here based on the MDN examples
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll